oracle/src/lib.rs

```diff
@@ -12,9 +12,11 @@ use ark_ff::Field;
 pub trait FqSponge<Fq: Field, G, Fr> {
     fn new(p: poseidon::ArithmeticSpongeParams<Fq>) -> Self;
     fn absorb_g(&mut self, g: &[G]);
+    fn absorb_fq(&mut self, x: &[Fq]);
     fn absorb_fr(&mut self, x: &[Fr]);
     fn challenge(&mut self) -> Fr;
     fn challenge_fq(&mut self) -> Fq;
 
     fn digest(self) -> Fr;
+    fn digest_fq(self) -> Fq;
```

Comment:
we should document all these functions btw

Reply:
Agreed, but not now and not here :)

Reply:
In general I'm on the board that if you're adding a new public function, you need to document it